src/shared/clients/iamClient.ts

```diff
@@ -71,6 +71,11 @@ export class DefaultIamClient {
 
     private getFriendlyName(arn: string): string {
         const tokens = arn.split('/')
+        if (tokens.length < 2) {
+            throw new Error(
+                `${arn} is an invalid IAM role ARN. Does not follow template arn:aws:iam::{id}/{friendlyName}.`
```

Comment:
Usually a good idea to put the variable part at the end of the message. It makes messages more predictable and familiar.

```suggestion
                `Invalid IAM role ARN (expected format: arn:aws:iam::{id}/{name}): ${arn}`
```